src/modules/rangeproof/main_impl.h

```diff
@@ -397,4 +397,105 @@ int secp256k1_rangeproof_verify_value(const secp256k1_context* ctx, const unsign
     return !memcmp(tmpch, &proof[offset], 32);
 }
 
+int secp256k1_rangeproof_create_value(const secp256k1_context* ctx, unsigned char* proof, size_t* plen, uint64_t value, const unsigned char* blind, const secp256k1_pedersen_commitment* commit, const secp256k1_generator* gen) {
+    secp256k1_ge commitp;
+    secp256k1_ge genp;
+    secp256k1_gej tmpj;
+    secp256k1_scalar es;
+    secp256k1_scalar tmps;
+    secp256k1_sha256 sha2;
+    unsigned char tmpch[33];
+    unsigned char pp_comm[32];
+    size_t offset;
+    size_t sz;
+    int overflow;
+
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(secp256k1_ecmult_gen_context_is_built(&ctx->ecmult_gen_ctx));
+    ARG_CHECK(proof != NULL);
+    ARG_CHECK(plen != NULL);
+    ARG_CHECK(blind != NULL);
+    ARG_CHECK(commit != NULL);
+    ARG_CHECK(gen != NULL);
+
+    if (*plen < 73 || (value == 0 && *plen < 65)) {
+        return 0;
```

Comment:
```
if (*plen < 65 || (value != 0 && *plen < 73)) {
```

or more readable (I think):
```
if (*plen < (value == 0) ? 65 : 73) {
```

(This implies that we should have also a test for `value == 0`.)

Reply:
BTW there is an explicit test for the size being correct when `value == 0`. You can search for the comment `DIfferent proof sizes are unfortunate`.

Reply:
Ah what I meant here is: If we had a (unit) test that calls this function with  `value == 0` and `65 <= size < 73`, it would have caught this bug.

Reply:
Lol, there's an even worse bug here -- we never actually set `plen`. It looks like I did basically no tests on `create_value`.